mitosheet/src/jupyter/notebook/extensionUtils.tsx

```diff
@@ -229,7 +229,16 @@ export const notebookWriteGeneratedCodeToCell = (analysisName: string, codeLines
 
     const codeCell = getCellAtIndex(mitosheetCallIndex + 1);
 
+
+    // We're removing the first line of the old code and the cell code because
+    // the cell code contains the analysis id and the old code does not
+    const oldCodeWithoutFirstLine = oldCode?.slice(1).join('\n');
+    const cellCodeWithoutFirstLine = getCellText(codeCell)?.split('\n').slice(1).join('\n');
```

Comment:
A more robust check might be: 
```
const endsWithOldCode = getCellText(codeCell).endsWith(oldCode)
```

And a comment explaining this as well

Reply:
You solution is better actually, given that it handles insertions more correctly. Def worth putting in a util in any case.